### PR TITLE
[receiver/hostmetrics] remove direction attribute

### LINKED ISSUE #11817: `[receiver/hostmetrics/networkscraper]`: remove `direction` attribute

### PR DESCRIPTION
**Pull request by codeboten**
The following change adds the direction to the metric name, removing the `direction` attribute.

Fixes #11817

### CODE REVIEW

**Comment by dmitryax** [comment]
I think this receiver is stable enough to avoid making such breaking changes. I believe many distributions depend on it. 

Can we make it through a feature gate (or enabling/disabling metrics) spanned across several versions in the following way?
0.55.0: keep reporting the old metrics with a warning that they will be replaced and provide a way to switch to the new metrics through a feature gate or disabling/enabling metrics.
0.57.0: switch to the new metrics with an option to bring the old metrics back.
0.59.0: remove old metrics.

**Comment by codeboten** [comment]
> reporting the old metrics with a warning

Would this warning show up at collector start?

**Review by dmitryax — COMMENTED** [review]
Overal LGTM

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml [hunk 1]
```diff
@@ -20,44 +20,124 @@ attributes:
...
-    description: The number of bytes transmitted and received.
+    description: The number of bytes transmitted and received. (Deprecated)
     unit: "By"
     sum:
       value_type: int
       aggregation: cumulative
       monotonic: true
     attributes: [device, direction]
 
+  system.network.io.transmit:
```
[the remark this answers is not in the document]
Comment (codeboten) [reply receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml:128]: Done 👍🏻

**Review by dmitryax — APPROVED** [review]

**Comment by sweetpotato0** [comment]
i have a quetion: why set the some metric to `Deprecated`, such as `system.network.dropped`、`system.network.errors`

**Comment by dmitryax** [comment]
@sweetpotato0 thanks for pointing on this. The deprecated labels should've been removed some time ago. I submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16227 to fix it

**Pull request merged**